docker/Dockerfile.iroh-gateway

```diff
@@ -0,0 +1,51 @@
+################################################################################
+## Builder
+################################################################################
+FROM rust:latest AS builder
+
+RUN update-ca-certificates
+
+# install latest protocol buffer compiler. Yes, it's really this irritating.
+# recent build URLs are missing "3" version prefix. version is actually "3.21.9"
+ENV RUST_BACKTRACE=1 \
+  PROTOC_VERSION=21.9 \
+  PROTOC_ZIP=protoc-21.9-linux-aarch_64.zip \
+  # PROTOC_ZIP=protoc-21.9-linux-x86_64.zip \
+  PROTOC=/usr/local/bin/protoc \
+  PROTOC_INCLUDE=/usr/local/include
+
+RUN curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v$PROTOC_VERSION/$PROTOC_ZIP \
+  && unzip -o $PROTOC_ZIP -d /usr/local bin/protoc \
+  && unzip -o $PROTOC_ZIP -d /usr/local 'include/*' \
+  && rm -f $PROTOC_ZIP \
+  && echo $($PROTOC --version)
+
+# has the side effect of updating the crates.io index & installing rust toolchain
+# called in a separate step for nicer caching. the command itself will fail,
+# b/c empty-library is not a dependency, so we override with an exit code 0
+RUN cargo install empty-library; exit 0
+
+WORKDIR /iroh
+
+COPY ../ .
+
+RUN cargo build --bin iroh-gateway --release
```

Comment:
Can we build with `cargo build --profile ci --bin iroh-gateway`. This uses the release profile we want to maintain for binaries (additional perf optimizations & slimmer binaries) and also makes future changes simple `.toml` changes instead of needing to touch all scripts.

Reply:
how about we do one-better & create a `docker` profile that matches `ci` so they can change independently?

Reply:
sgtm, I can push that update then

Reply:
pretty sure I can handle in this PR, it's just a change to the workspace `Cargo.toml`, right?